layout-service/src/main/java/io/meeds/layout/service/PageLayoutService.java

```diff
@@ -291,8 +291,13 @@ public void updatePageApplicationPreferences(PageKey pageKey,
                                                       applicationId,
                                                       pageKey));
     }
-    Portlet portletPreferences = new Portlet();
-    preferences.forEach(preference -> portletPreferences.setValue(preference.getName(), preference.getValue()));
+    Portlet portletPreferences = layoutService.load(application.getState());
+    if (portletPreferences == null) {
+      portletPreferences = new Portlet();
+    }
+    for (PortletInstancePreference preference : preferences) {
+      portletPreferences.setValue(preference.getName(), preference.getValue());
+    }
```

Comment:
Please keep it as is, since it must delete existing Portlet preferences if not set in the Application Update Request.

```suggestion
    Portlet portletPreferences = new Portlet();
    preferences.forEach(preference -> portletPreferences.setValue(preference.getName(), preference.getValue()));
```

Reply:
Ok, I rollback this point. I don't remember why I change it.

Reply:
just commited

Reply:
I now remember why I change this. 

In my portlets, I call /layout/rest/pages/application/preferences?pageRef=portal::global::login&applicationId=449 to set portletPrefs for the appId 449.

Without my modification, PageLayoutRest.updatePageApplicationPreferences do not really make an update. 
it completly replace existing pref by new one. 

I'm not sure to understand why 

>  it must delete existing Portlet preferences if not set in the Application Update Request.

should we have separated endpoint for update and replace ?

Reply:
Yes, or if you want, to keep the current behavior compatible with the newly desired one, add a parameter `replaceExisting = true` by default (or `keepExisting = false`)

Reply:
option keepExisting added, with default value=false

done